src/components/AddPlayerForm/AddPlayerForm.js

```diff
@@ -0,0 +1,45 @@
+import styled from 'styled-components/macro'
+import Button from '../Button'
+
+export default function AddPlayerForm({
+  onAddPlayer,
+  disabled,
+  placeholderText,
+}) {
+  return (
+    <Form onSubmit={handleSubmit}>
+      <label>
+        Add Player:
+        <input
+          name="playerName"
+          required
+          placeholder={placeholderText}
+          minLength="2"
+          maxLength="30"
+          disabled={disabled}
+          title="addPlayerInput"
+        ></input>
+      </label>
+      <Button disabled={disabled} title="addPlayerButton">
+        Add
+      </Button>
+    </Form>
+  )
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    const form = event.target
+    const { playerName } = form.elements
+
+    onAddPlayer({
+      nameOfPlayer: playerName.value,
+    })
+    form.reset()
+    playerName.focus()
```

Comment:
I wonder what happens here after the 4th player is submitted 🤔 Will the focus just get ignored or does it throw an error?

Reply:
the input field is getting disabled and the placeholder message changes to inform you that the maximum of four players has reached... 😄

Reply:
the input field is getting disabled and the placeholder text changes to inform you, that the max of 4 players is reached...